src/internal/connector/onedrive/collection.go

```diff
@@ -252,22 +300,49 @@ func (oc *Collection) populateItems(ctx context.Context) {
 				itemSize = itemInfo.OneDrive.Size
 			}
 
-			itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
-				progReader, closer := observe.ItemProgress(ctx, itemData, observe.ItemBackupMsg, itemName, itemSize)
-				go closer()
-				return progReader, nil
-			})
+			if isFile {
+				itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
+					progReader, closer := observe.ItemProgress(ctx, itemData, observe.ItemBackupMsg, itemName+DataFileSuffix, itemSize)
+					go closer()
+					return progReader, nil
+				})
+
+				oc.data <- &Item{
+					id:   itemName + DataFileSuffix,
+					data: itemReader,
+					info: itemInfo,
+				}
+			}
+
+			if itemMeta != nil {
+				// TODO(meain): This right now creates two
```

Comment:
Should we have a separate progressbar entry for upload metadata? How should that behave?

Reply:
If possible, we should treat it like any other item in the collection.

Reply:
We are good here then. As of now we do treat them as just another item in the collection. You will get one progressbar when upload the data and one when uploading the metadata.